src/app/story/directives/hero.component.ts

```diff
@@ -93,7 +111,51 @@ export class StoryHeroComponent implements OnInit, OnChanges {
     this.story.discard();
   }
 
+  get pendingPublishedAt(): Date {
+    if (!this.story.publishedAt && !this._pendingPublishedAt) {
+      // really only want to set this once and story.publishedAt not ready when instantiated
+      this._pendingPublishedAt = new Date();
+    }
+    if (this.story.publishedAt && typeof this.story.publishedAt === "string") {
+      // story.publishedAt supposed to be a Date but it's actually a string ಠ_ಠ
+      // so assigning it to the Date value so it doesn't have to parse the string to a Date over and over and over again
+      this.story.publishedAt = new Date(this.story.publishedAt);
+      return this.story.publishedAt;
+    } else if (this.story.publishedAt) {
+      return this.story.publishedAt;
+    } else {
+      return this._pendingPublishedAt;
+    }
+  }
+
+  set pendingPublishedAt(date: Date) {
+    this._pendingPublishedAt = date;
+  }
+
+  get pendingPublishedAtHour(): string {
+    if (this.pendingPublishedAt.getMinutes() <= 30) {
+      return '' + this.pendingPublishedAt.getHours();
+    } else {
+      return this.pendingPublishedAt.getHours() + 1 <= 23 ? '' + (this.pendingPublishedAt.getHours() + 1) : '00';
+    }
+  }
+
+  set pendingPublishedAtHour(hour: string) {
+    this._pendingPublishedAt.setHours(Number(hour));// TODO: timezone considerations
+  }
+
+  get pendingPublishedAtMinutes(): string {
+    return this.pendingPublishedAt.getMinutes() <= 30 ? '30' : '00';
+  }
+
+  set pendingPublishedAtMinutes(minutes: string) {
+    this._pendingPublishedAt.setMinutes(Number(minutes));
+  }
+
   togglePublish() {
+    if (!this.story.publishedAt) {
+      this.story.set('publishedAt', this._pendingPublishedAt);
```

Comment:
This is a bad deal because it means the story has changed(). Not sure what the pattern should be to set and persist this value. But I think we first need to resolve what the requirements are for #154.

(Also not sure if the user can set publishedAt when the story has already been published or if dates in the past are allowed.)